packages/transaction-controller/src/utils/validation.ts

```diff
@@ -64,6 +65,25 @@ export function validateTxParams(
   validateGasFeeParams(txParams);
 }
 
+/**
+ * Validates type property, ensuring that if it is specified, it is a valid transaction envelope type.
+ *
+ * @param type - The transaction envelope type to validate.
+ * @throws Throws invalid params if the type is not a valid transaction envelope type.
+ */
+function validateEnvelopeType(type: string | undefined) {
```

Comment:
I think this is a good addition, but for clarity, I believe we already do this partially via `ensureProperTransactionEnvelopeTypeProvided` as we throw if `maxFeePerGas` or `maxPriorityFeePerGas` is set and `type` isn't fee market, but if `gasPrice` is set, we only throw if `type` is fee market.

So this should add consistency for the cases where no gas is provided.